app/src/main/java/com/pvp/app/ui/common/Components.kt

```diff
@@ -208,4 +220,204 @@ fun TextField(
         },
         messages = errors
     )
+}
+
+@Composable
+fun ButtonWithDialog(
+    modifier: Modifier = Modifier,
+    contentAlignment: Alignment = Alignment.TopStart,
+    mainButtonContent: @Composable RowScope.() -> Unit = { Text("Open Dialog") },
+    confirmButtonContent: @Composable RowScope.() -> Unit = { Text("Confirm") },
+    dismissButtonContent: @Composable RowScope.() -> Unit = { Text("Dismiss") },
+    dialogTitle: @Composable () -> Unit = { Text("Dialog Title") },
+    dialogContent: @Composable () -> Unit = { Text("Dialog Content") },
+    onConfirmClick: () -> Unit = {},
+    onDismiss: () -> Unit = {}
+) {
+    var showDialog by remember { mutableStateOf(false) }
+
+    Box(
+        modifier = modifier,
+        contentAlignment = contentAlignment
+    ) {
+        androidx.compose.material3.Button(
+            content = mainButtonContent,
+            onClick = {
+                showDialog = true
+            }
+        )
+    }
+
+    if (showDialog) {
+        AlertDialog(
+            title = dialogTitle,
+            text = dialogContent,
+            onDismissRequest = {
+                onDismiss()
+
+                showDialog = false
+            },
+            confirmButton = {
+                Box(
+                    contentAlignment = Alignment.BottomEnd
+                ) {
+                    androidx.compose.material3.Button(
+                        content = confirmButtonContent,
+                        onClick = {
+                            onConfirmClick()
+
+                            showDialog = false
+                        }
+                    )
+                }
+            },
+            dismissButton = {
+                Box(
+                    contentAlignment = Alignment.BottomEnd
+                ) {
+                    androidx.compose.material3.Button(
+                        content = dismissButtonContent,
+                        onClick = {
+                            onDismiss()
+
+                            showDialog = false
+                        },
+                    )
+                }
+            }
+        )
+    }
+}
+
+@Composable
+fun IconButtonWithDialog(
+    modifier: Modifier = Modifier,
+    icon: ImageVector = Icons.Outlined.Edit,
```

Comment:
Seems weird to have a default icon which is for `edit` purpose in a component that must use icon. Find a better replacement or just make it a required parameter. Don't forget to modify any usage of this component so that no side-effects appear.